### PR TITLE
Add compilerEnvMap for managing compiler env

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit adds compilerEnvMap for managing compiler environment
with backend name.

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

**Comment by jyoungyun** [comment]
This PR depends on #537 PR.

**Comment by jyoungyun** [comment]
@YongseopKim 

Could you review this PR? I introduce gCompilerEnv map interface to managing compile environment with backend name.

**Review by YongseopKim — APPROVED** [review]
LGTM

**Comment by YongseopKim** [comment]
So, can I close https://github.com/Samsung/ONE-vscode/pull/537?

**Comment by YongseopKim** [comment]
Oh now I see `no merge`. What does it mean?

**Comment by jyoungyun** [comment]
> Oh now I see no merge. What does it mean?

This PR has a dependency with #537 PR.
I made a separated PR for this one. So #537 need to be merged first.
The reason why I ask you for review is that I want your confirm of this PR first. If it is not matched with your approach about API, I need to change it ASAP.

**Comment by YongseopKim** [comment]
I like this approach. I hope #537 is merged ASAP!

**Comment by jyoungyun** [comment]
Ready to review :)
Please take another look!
/cc @YongseopKim @hyunsik-yoon

**Review by YongseopKim — APPROVED** [review]
LGTM

File: src/Backend/Backend.ts [hunk 1]
```diff
@@ -34,6 +36,10 @@ function backendRegistrationApi() {
       const backendName = backend.name();
       assert(backendName.length > 0);
       globalBackendMap[backendName] = backend;
+      const compiler = backend.compiler();
```
Comment (jyoungyun) [inline src/Backend/Backend.ts:39]: I will change this `compiler` name to `toolchain` on next PR.
There are some patches implemented using this name, so I want to apply this code first and change the name later.

Comment (YongseopKim) [reply src/Backend/Backend.ts:39]: 👍

**Review by hyunsik-yoon — APPROVED** [review]

**Pull request merged**